web/config/app.go

```diff
@@ -0,0 +1,98 @@
+package config
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"strconv"
+
+	"github.com/beatlabs/github-auth/app"
+	"github.com/beatlabs/github-auth/key"
+	gh "github.com/google/go-github/v35/github"
+)
+
+// GithubAppConfig keeps parameters of the GitHub app
+type GithubAppConfig struct {
+	AppID    string
+	ClientID string
+	Secret   string
+}
+
+// TODO: rename because confusing
+type GithubApp struct {
+	Config *GithubAppConfig
+	App    *app.Config
+}
+
+func NewAppConfig() *GithubAppConfig {
+	return &GithubAppConfig{
+		AppID:    os.Getenv(AppEnv),
+		ClientID: os.Getenv(KeyEnv),
+		Secret:   os.Getenv(SecretEnv),
+	}
+}
+
+func (conf *GithubAppConfig) Valid() bool {
+	return conf.AppID != "" &&
+		conf.ClientID != "" && conf.Secret != ""
+}
+
+// AppClient creates client for the Quickfeed GitHub Application
+// This client can only access the metadata of the Application itself
+// To access organizations via GitHub API we need to derive an installation client
+// from this Application client for each course organization
+func NewApp() (*GithubApp, error) {
+	config := NewAppConfig()
+	if !config.Valid() {
+		return nil, fmt.Errorf("Error configuring GitHub App: %+v", config)
+	}
+	appKey, err := key.FromFile(appKeyPath)
```

Comment:
Do we need to read the key material from file, or can it be read from ENV var? I see it is currently a pem file and not a token... so perhaps it is necessary.

Reply:
GitHub wants a pem file as an app key, but I think that a path to the file should be read from ENV.